tracer/src/Datadog.Trace/Sampling/CustomSamplingRule.cs

```diff
@@ -30,8 +30,20 @@ internal class CustomSamplingRule : ISamplingRule
             string operationNameRegex)
         {
             _samplingRate = rate;
-            _serviceNameRegex = WrapWithLineCharacters(serviceNameRegex);
-            _operationNameRegex = WrapWithLineCharacters(operationNameRegex);
+
+            _serviceNameRegex = serviceNameRegex is null
+                                    ? null
+                                    : new(
+                                        WrapWithLineCharacters(serviceNameRegex),
+                                        RegexOptions.Compiled,
+                                        RegexTimeout);
+            _operationNameRegex = operationNameRegex is null
+                                      ? null
+                                      : new(
+                                          WrapWithLineCharacters(operationNameRegex),
```

Comment:
can a new Regex() throw if the pattern isn't right? Shall we out this in a try catch?

Reply:
Good question, I don't know, so I'll test that out

Reply:
Pushed up a change for this in 77a392921dcf3b0a0eb4868a7cf2bb36d036870c

I left the ctor throwing so that the `BuildFromConfigurationString` would catch the exception and return no rules.
This might be a change of behavior from before where previously I think invalid regex would've thrown in the `Regex.IsMatch` call.
Unsure really which is better.

Reply:
Throwing an exception (and catching it / logging it) at configuration seems like an improvement!

Reply:
TBH I hadn't seen that there were a try catch `BuildFromConfigurationString`, I was afraid we would throw at that time and not handling it whereas we used to throw later and handled it. So my main concern was already handled. But what you've done is an improvement. Thanks